plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1391 @@
+    {
+        if (!hasPermission(createResource(functionName), context, TrinoAccessType.CREATE, "CreateFunction")) {
+            AccessDeniedException.denyCreateFunction(functionName.toString());
+        }
+    }
+
+    @Override
+    public void checkCanDropFunction(SystemSecurityContext context, CatalogSchemaRoutineName functionName)
+    {
+        if (!hasPermission(createResource(functionName), context, TrinoAccessType.DROP, "DropFunction")) {
+            AccessDeniedException.denyDropFunction(functionName.toString());
+        }
+    }
+
+    @Override
+    public void checkCanShowCreateFunction(SystemSecurityContext context, CatalogSchemaRoutineName functionName)
+    {
+        if (!hasPermission(createResource(functionName), context, TrinoAccessType.SHOW, "ShowCreateFunction")) {
+            AccessDeniedException.denyShowCreateFunction(functionName.toString());
+        }
+    }
+
+    @Override
+    public void checkCanShowFunctions(SystemSecurityContext context, CatalogSchemaName schema)
+    {
+        if (!hasPermission(createResource(schema), context, TrinoAccessType.SHOW, "ShowFunctions")) {
+            AccessDeniedException.denyShowFunctions(schema.toString());
+        }
+    }
+
+    @Override
+    public boolean canExecuteFunction(SystemSecurityContext context, CatalogSchemaRoutineName functionName)
+    {
+        return hasPermission(createResource(functionName), context, TrinoAccessType.EXECUTE, "ExecuteFunction");
+    }
+
+    @Override
+    public boolean canCreateViewWithExecuteFunction(SystemSecurityContext context, CatalogSchemaRoutineName functionName)
+    {
+        return hasPermission(createResource(functionName), context, TrinoAccessType.CREATE, "CreateViewWithExecuteFunction");
+    }
+
+    @Override
+    public Set<SchemaFunctionName> filterFunctions(SystemSecurityContext context, String catalogName, Set<SchemaFunctionName> functionNames)
+    {
+        Set<SchemaFunctionName> toExclude = null;
+
+        for (SchemaFunctionName functionName : functionNames) {
+            RangerTrinoResource res = createResource(catalogName, functionName);
+
+            if (!hasPermissionForFilter(res, context, TrinoAccessType._ANY, "filterFunctions")) {
+                LOG.debug("filterFunctions(user={}): skipping function {}.{}.{}", context.getIdentity(), catalogName, functionName.getSchemaName(), functionName.getFunctionName());
+
+                if (toExclude == null) {
+                    toExclude = new HashSet<>();
+                }
+
+                toExclude.add(functionName);
+            }
+        }
+
+        return toExclude == null ? functionNames : functionNames.stream().filter(((Predicate<? super SchemaFunctionName>) toExclude::contains).negate()).collect(Collectors.toSet());
+    }
+
+    @Override
+    public List<ViewExpression> getRowFilters(SystemSecurityContext context, CatalogSchemaTableName tableName)
```

Comment:
Do we have tests for row filters and column masks?

Reply:
Yes, `TestRangerSystemAccessControl.testRowFilters()` and `TestRangerSystemAccessControl.testColumnMask()` cover these.